packages/block-editor/src/store/selectors.js

```diff
@@ -1284,6 +1284,45 @@ export const hasInserterItems = createSelector(
 	],
 );
 
+/**
+ * Determines whether there is only one item that may be inserted.
+ *
+ * @param {Object}  state        Editor state.
+ * @param {?string} rootClientId Optional root client ID of block list.
+ *
+ * @return {boolean} True if there is one item available, false if zero or more than one.
+ */
+export const __experimentalHasOnlyOneAllowedBlockType = ( state, rootClientId = null ) => {
+	if ( rootClientId ) {
+		const parentBlockListSettings = getBlockListSettings( state, rootClientId );
+		return get( parentBlockListSettings, [ 'allowedBlocks', 'length' ], 0 ) === 1;
```

Comment:
I found it interesting that the `hasInserterItems` selector works quite differently in determining blocks available for insert. I wonder what the reason behind that is.

Reply:
Specifically it does this:
```
const hasBlockType = some(
	getBlockTypes(),
	( blockType ) => canIncludeBlockTypeInInserter( state, blockType, rootClientId )
);
```

Reply:
this selector is gone now